fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateRoutineLoadAnalyzer.java

```diff
@@ -0,0 +1,50 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+package com.starrocks.sql.analyzer;
+
+import com.google.common.base.Strings;
+import com.starrocks.analysis.CreateRoutineLoadStmt;
+import com.starrocks.analysis.LabelName;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.common.FeNameFormat;
+import com.starrocks.common.UserException;
+import com.starrocks.qe.ConnectContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class CreateRoutineLoadAnalyzer {
+
+    private static final Logger LOG = LogManager.getLogger(CreateRoutineLoadAnalyzer.class);
+    private static final String NAME_TYPE = "ROUTINE LOAD NAME";
+
+    private CreateRoutineLoadAnalyzer() {
+        throw new IllegalStateException("creating an instance is illegal");
+    }
+
+    public static void analyze(CreateRoutineLoadStmt statement, ConnectContext context) {
+        LabelName label = statement.getLabelName();
+        String dbName = label.getDbName();
+        if (Strings.isNullOrEmpty(dbName)) {
+            dbName = context.getDatabase();
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+        }
+        label.setDbName(dbName);
```

Comment:
why set DbName here?

Reply:
better to move it in if branch?

Reply:
like this lable's dbname is null. so, need set dbname
CREATE ROUTINE LOAD labelname ON tablename......
lilke this lable's dbname is ok
CREATE ROUTINE LOAD dbname.labelname ON tablename......

[move setdb into if branch](https://github.com/StarRocks/starrocks/pull/9981/commits/7e6f29fd32b99690754d473ea58921be91175b73)